raiden/network/rpc/client.py

```diff
@@ -540,27 +457,26 @@ def send_transaction(
                 'behavior.'
             )
 
-        if to == b'0' * 40:
+        if to == b'0' * 20:
             warnings.warn('For contract creation the empty string must be used.')
 
-        nonce = self.nonce()
-        startgas = self.check_startgas(startgas)
-
-        tx = Transaction(
-            nonce,
-            self.gasprice(),
-            startgas,
-            to=to,
+        transaction = dict(
+            nonce=self.nonce(),
+            gasPrice=self.gasprice(),
+            gas=self.check_startgas(startgas),
             value=value,
-            data=data,
+            data=data
         )
 
-        tx.sign(self.privkey)
-        result = self.rpccall_with_retry(
-            'eth_sendRawTransaction',
-            data_encoder(rlp.encode(tx)),
-        )
-        return result[2 if result.startswith('0x') else 0:]
+        # add the to address if not deploying a contract
+        if to != b'':
+            transaction['to'] = to_checksum_address(to)
+
+        signed_txn = self.web3.eth.account.signTransaction(transaction, self.privkey)
+
+        result = self.web3.eth.sendRawTransaction(signed_txn.rawTransaction)
+        encoded_result = encode_hex(result)
+        return encoded_result[2 if encoded_result.startswith('0x') else 0:]
```

Comment:
I think this is will expand to either `[2:]` or `[0:]`

Reply:
No, it's right (get from the second byte if the string starts with `0x`, from byte zero/everything otherwhise). But I'd recommend using eth_utils' `remove_0x_prefix` instead of coming with a custom implementation everytime.

Reply:
damn you are right. What a weird syntax. I'd expect it returns `[2]`
```
x = '0xabcdef'
>>> x[2 if x[0:2] == '0x' else 0:]
'abcdef'
```